### ISSUE TITLE
positive false for lint/correctnewss/noUnusedVariables when dealing with identifier with unicode escape sequence

### ISSUE THREAD

**Issue by hardfist** [biomejs/biome#7479]
### Environment information

https://biomejs.dev/playground/?code=dgBhAHIAIABcAHUAMAAwADYANQAgAD0AIAAxADAAOwAKAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoAGUAKQA7AA%3D%3D
* input.js
```js
var \u0065 = 10;
console.log(e);
```

### What happened?

report unused_variable lint warning for `\0065`

### Expected result

not report lint warning like eslint 

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
